ocaml/xapi/message_forwarding.ml

```diff
@@ -6429,4 +6446,113 @@ functor
               ~base_build_id ~base_version ~base_release ~to_version ~to_release
         )
     end
+
+    module Observer = struct
+      module RefSet = Set.Make (struct
+        type t = [`host] Ref.t
+
+        let compare = Ref.compare
+      end)
+
+      let create ~__context ~name_label ~name_description ~hosts ~attributes
+          ~endpoints ~components ~enabled =
+        info "Observer.create: name_label=%s" name_label ;
+        Local.Observer.create ~__context ~name_label ~name_description ~hosts
+          ~attributes ~endpoints ~components ~enabled
+
+      let register ~__context ~self ~host =
+        info "Observer.unregister: self=%s" (observer_uuid ~__context self) ;
+        let local_fn = Local.Observer.register ~self ~host in
+        let client_fn session_id rpc =
+          Client.Observer.register ~rpc ~session_id ~self ~host
+        in
+        do_op_on ~__context ~host ~local_fn client_fn
+
+      let unregister ~__context ~self ~host =
+        info "Observer.unregister: self=%s" (observer_uuid ~__context self) ;
+        let local_fn = Local.Observer.unregister ~self ~host in
+        let client_fn session_id rpc =
+          Client.Observer.unregister ~rpc ~session_id ~self ~host
+        in
+        do_op_on ~__context ~host ~local_fn client_fn
+
+      let destroy ~__context ~self =
+        info "Observer.destroy: self=%s" (observer_uuid ~__context self) ;
+        Local.Observer.destroy ~__context ~self
+
+      let set_hosts ~__context ~self ~value =
```

Comment:
With the split logic using register/unregister we can move this logic into `xapi_tracing.ml` more direclty.

Reply:
This has been done